.github/workflows/build.yml

```diff
@@ -7,46 +7,60 @@ on:
       # Push events on the main branch
       - main
       - release/**
+      - set-product-version
 
 env:
   PKG_NAME: consul
+  VERSION: oss
 
 jobs:
-  get-product-version:
+  set-product-version:
     runs-on: ubuntu-latest
     outputs:
-      product-version: ${{ steps.get-product-version.outputs.product-version }}
+      product-version: ${{ steps.set-product-version.outputs.product-version }}
+      base-product-version: ${{ steps.set-product-version.outputs.base-product-version }}
       product-date: ${{ steps.get-product-version.outputs.product-date }}
-      pre-version: ${{ steps.get-product-version.outputs.pre-version }}
+      pre-version: ${{ steps.set-product-version.outputs.prerelease-product-version }}
       pkg-version: ${{ steps.get-product-version.outputs.pkg-version }}
       shared-ldflags: ${{ steps.shared-ldflags.outputs.shared-ldflags }}
     steps:
       - uses: actions/checkout@v2
+      - name: set product version
+        id: set-product-version
+        uses: hashicorp/actions-set-product-version@v1
       - name: get product version
         id: get-product-version
         run: |
-          CONSUL_VERSION=$(build-support/scripts/version.sh -r)
           CONSUL_DATE=$(build-support/scripts/build-date.sh)
           ## TODO: This assumes `make version` outputs 1.1.1+ent-prerel
-          IFS="+" read VERSION _other <<< "$CONSUL_VERSION"
-          IFS="-" read _other PREREL_VERSION <<< "$CONSUL_VERSION"
-          ## TODO: this assumes `version.sh` outputs in the expected ordering of
-          ##       [version]+ent{-prerelease}  If we need to transition to
-          ##       [version]{-prerelease}+ent before then, we'll need to add
-          ##       logic to handle presense/absence of the prerelease
-          echo "::set-output name=product-version::${CONSUL_VERSION}"
           echo "::set-output name=product-date::${CONSUL_DATE}"
-          echo "::set-output name=pre-version::${PREREL_VERSION}"
-          echo "::set-output name=pkg-version::${VERSION}"
+          echo "::set-output name=pkg-version::${{ env.VERSION }}"
 
       - name: Set shared -ldflags
         id: shared-ldflags
         run: |
           T="github.com/hashicorp/consul/version"
-          echo "::set-output name=shared-ldflags::-X ${T}.GitCommit=${GITHUB_SHA::8} -X ${T}.GitDescribe=${{ steps.get-product-version.outputs.product-version }} -X ${T}.BuildDate=${{ steps.get-product-version.outputs.product-date }}"
-
+          echo "::set-output name=shared-ldflags::-X ${T}.GitCommit=${GITHUB_SHA::8} \
+          -X ${T}.GitDescribe=${{ steps.set-product-version.outputs.product-version }} \
+          -X ${T}.BuildDate=${{ steps.get-product-version.outputs.product-date }} \
+          -X ${T}.Version=${{ steps.set-product-version.outputs.base-product-version }} \
```

Comment:
we can drop setting the Version, VersionPrerelease, VersionMetadata here I think since we're using go:embed - like we did in the consul-enterprise PR -